src/frontend/src/binder/expr/function.rs

```diff
@@ -1012,6 +1012,26 @@ impl Binder {
                 ("to_ascii", raw_call(ExprType::ToAscii)),
                 ("to_hex", raw_call(ExprType::ToHex)),
                 ("quote_ident", raw_call(ExprType::QuoteIdent)),
+                ("quote_literal", guard_by_len(1, raw(|_binder, mut inputs| {
+                    if inputs[0].return_type() != DataType::Varchar {
+                        // Support `quote_literal(any)` by converting it to `quote_literal(any::text)`
+                        // Ref. https://github.com/postgres/postgres/blob/REL_16_1/src/include/catalog/pg_proc.dat#L4641
+                        FunctionCall::cast_mut(&mut inputs[0], DataType::Varchar, CastContext::Explicit).map_err(|e| {
+                            ErrorCode::BindError(format!("{} in `quote_literal`", e.as_report()))
```

Comment:
It is an anti-pattern to interpolate source error (or its report just to bypass the lint) into error message. The better way is to pass it structurally as `ErrorCode::BindErrorRoot`'s field `error` (annotated with `#[source]`). This is the general principle. cc @BugenZhao CMIIAW

Specific to this case, it may be enough to simply `?` as the only new context added here is function name `quote_literal`. It is already available in the root error (refer to example below), or can be added in the caller.

(The only data type not allowed to cast to text in today's risingwave is the internal `serial`)
```
dev=> create table t (v1 int);
CREATE_TABLE

dev=> select 1+quote_literal(_row_id) from t;
ERROR:  Failed to run the query

Caused by these errors (recent errors listed first):
  1: Failed to bind expression: 1 + quote_literal(_row_id)
  2: Bind error: cannot cast type "serial" to "character varying" in Explicit context
```

Reply:
Simply `?` looks good. Updated